src/protocols/router/hpr_router_relay.erl

```diff
@@ -0,0 +1,201 @@
+-module(hpr_router_relay).
+
+-behaviour(gen_server).
+
+% API
+-export([
+    start/2
+]).
+
+% gen_server callbacks
+-export([
+    init/1,
+    handle_continue/2,
+    handle_call/3,
+    handle_cast/2
+]).
+
+-record(state, {
+    monitor_process :: pid(),
+    gateway_stream :: hpr_router_stream_manager:gateway_stream(),
+    router_stream :: grpc_client:client_stream()
+}).
+
+% ------------------------------------------------------------------------------
+% API
+% ------------------------------------------------------------------------------
+
+-spec start(
+    hpr_router_stream_manager:gateway_stream(),
+    grpc_client:client_stream()
+) -> {ok, pid()}.
+%% @doc Start this service.
+start(GatewayStream, RouterStream) ->
+    gen_server:start(?MODULE, [GatewayStream, RouterStream], []).
+
+% ------------------------------------------------------------------------------
+% gen_server callbacks
+% ------------------------------------------------------------------------------
+
+-spec init(list()) -> {ok, #state{}, {continue, relay}}.
+init([GatewayStream, RouterStream]) ->
+    {ok, MonitorPid} =
+        hpr_router_relay_monitor:start(
+            self(), GatewayStream, RouterStream
+        ),
+    {
+        ok,
+        #state{
+            monitor_process = MonitorPid,
+            gateway_stream = GatewayStream,
+            router_stream = RouterStream
+        },
+        {continue, relay}
+    }.
+
+-spec handle_continue(relay, #state{}) -> {noreply, #state{}, {continue, relay}}.
+handle_continue(relay, State) ->
+    handle_rcv_response(grpc_client:rcv(State#state.router_stream), State).
```

Comment:
Could we make `handle_rcv_response` just a case clause in here? The module seems simple enough.
Also spec is is missing the `{stop, ..., ...,}`

Reply:
done.